pkg/metricsprovider/signalfx.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2020 PayPal
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metricsprovider
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"metrics/loadwatcher/pkg/watcher"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// SignalFX Request Params
+	signalFxBaseUrl        = "https://api.pypl-us0.signalfx.com/v1/timeserieswindow"
```

Comment:
It would be nice to automatically grab the endpoint.

Reply:
It would depend on the users - some may need to provide it statically and others may lookup hostname from a service registry, depending on the user setup. Since looking it up is client side dependent, we don't have control over that code.

Reply:
We can however make it configurable (via env var etc.).